src/common/config.go

```diff
@@ -305,6 +355,64 @@ func SandboxConfJson() string {
 	return string(s)
 }
 
+// SaveConfigAtomic saves a config to a file atomically to prevent race conditions
+// This is used for template.json creation to avoid corruption during concurrent access
+func SaveConfigAtomic(cfg *Config, filePath string) error {
+	// Create temp file in same directory as target file
+	dir := filepath.Dir(filePath)
+	tempFile, err := os.CreateTemp(dir, filepath.Base(filePath)+".tmp.*")
```

Comment:
rename is frequently used to atomically update a file, but I'm not sure it will work correctly if we're renaming across different file systems.  I'm not sure the temp file will always be on the same FS as the final final.  Perhaps we should write to filePath + ".tmp" so that the write goes to the same dir/FS.